### PR TITLE
Adding the implementation for WinRM enablement feature in AzureRG

### PR DESCRIPTION
**Pull request by roniga**

### CODE REVIEW

File: Tasks/DeployAzureResourceGroup/task.json [hunk 1]
```diff
@@ -116,6 +116,24 @@
             "visibleRule": "action = Create Or Update Resource Group"
         },
         {
+            "name": "enableRemoteDeploymentForCreate",
+            "type": "boolean",
+            "label": "Enable Remote Deployment",
+            "defaultValue": "true",
+            "visibleRule": "action = Create Or Update Resource Group",
```
Comment (kmkumaran) [inline Tasks/DeployAzureResourceGroup/task.json]: Does VisibleRule have an Or notion where you can specify both create and select actions?

File: Tasks/DeployAzureResourceGroup/Utility.ps1 [hunk 2]
```diff
@@ -752,4 +755,154 @@ function Get-AzureRMVMsConnectionDetailsInResourceGroup
...
+}
+
+function Add-AzureVMCustomScriptExtension
+{
+    param([string]$resourceGroupName,
+          [object]$vmResource)
+	
+    $vmName = $vmResource.Name
+    $location=$vmResource.Location	
+    $dnsName=[string]::Concat("*.",$location,".cloudapp.azure.com")
```
Comment (kmkumaran) [inline Tasks/DeployAzureResourceGroup/Utility.ps1]: Should we hard code the suffix? Is it possible to get that from resource property?

**Comment by pryenugu** [comment]
Looks good.

File: Tasks/DeployAzureResourceGroup/task.json (hunk 1 above)
Comment (pavanadepu2) [inline Tasks/DeployAzureResourceGroup/task.json]: No, currently there is no support for logical ops.

File: Tasks/DeployAzureResourceGroup/Utility.ps1 (hunk 2 above)
Comment (pavanadepu2) [inline Tasks/DeployAzureResourceGroup/Utility.ps1]: The suffix is fixed for all azure vms. It is a non-editable field even in azure portal.

**Pull request merged**